### DISCUSSION TITLE
os.findlib Windows bug?

### DISCUSSION THREAD

**Discussion by namiestew** [premake/premake-core#2210] [General]
I am not too sure if this is a bug but I found that the additional paths passed to the _os.findlib_ function is not being concatenated with the correct delimiter for Windows which results in nothing being found.

As can be seen in line 149 and line 154 of _os.lua_ file, the paths are being delimited with a colon instead of a semicolon for Windows. This will result in invalid paths being used in _os.pathsearch_ and no results would get returned. Is someone able to confirm if my observation is correct? 

![image](https://github.com/premake/premake-core/assets/102791593/e5c9742d-467a-4a2c-846f-29c8dea644e2)

**Comment by criptych** [comment]
I thought this was odd, too, especially since the rest of Premake seems to be so VS- (and thus Windows-) focused.  At first, I thought it's because this is only an internal path string, and the distinction would be handled within `os.pathsearch`; but there it specifically *ignores* `:` on Windows because it's a drive separator.  

But! if you don't specify additional paths, everything works fine, because the `PATH` environment variable is essentially passed through (from `get_library_search_path`) and used unchanged.

**Comment by Jarod42** [comment]
Should be fixed by https://github.com/premake/premake-core/pull/2442
